test/integration/deep/opaqueports/opaque_ports_test.go

```diff
@@ -14,171 +15,281 @@ import (
-					return fmt.Errorf("failed to find expected TCP metric when pod is opaque\n%s", metrics)
-				}
-
-				return nil
 			})
+		}
+	})
+}
 
-			if err != nil {
-				testutil.AnnotatedFatalf(t, "unexpected metric output", "unexpected metric output: %s", err)
-			}
-		})
-
-		t.Run("expect inbound TCP connection metric with expected TLS identity for opaque service app", func(t *testing.T) {
-			// Check the slow cooker metrics
-			err := TestHelper.RetryFor(30*time.Second, func() error {
-				pods, err := TestHelper.GetPods(ctx, opaquePortsNs, map[string]string{"app": opaqueSvcSC})
-				if err != nil || len(pods) == 0 {
-					return fmt.Errorf("error getting pods\n%w", err)
-				}
-				metrics, err := getPodMetrics(pods[0], opaquePortsNs)
-				if err != nil {
-					return fmt.Errorf("error getting metrics for pod\n%w", err)
-				}
-				if httpRequestTotalMetricRE.MatchString(metrics) {
-					return fmt.Errorf("expected not to find HTTP outbound requests when service is opaque\n%s", metrics)
-				}
-				// Check the application metrics
-				pods, err = TestHelper.GetPods(ctx, opaquePortsNs, map[string]string{"app": opaqueSvcApp})
-				if err != nil {
-					return fmt.Errorf("error getting pods\n%w", err)
-				}
-				metrics, err = getPodMetrics(pods[0], opaquePortsNs)
-				if err != nil {
-					return fmt.Errorf("error getting metrics for pod\n%w", err)
-				}
-				if !tcpMetricRE.MatchString(metrics) {
-					return fmt.Errorf("failed to find expected TCP metric when pod is opaque\n%s", metrics)
-				}
+func TestOpaquePortsCalledByPodTarget(t *testing.T) {
+	ctx := context.Background()
+	TestHelper.WithDataPlaneNamespace(ctx, "opaque-ports-called-by-pod-ip-test", map[string]string{}, t, func(t *testing.T, opaquePortsNs string) {
+		checks := func(c ...check) []check { return c }
 
-				return nil
-			})
+		if err := deployApplications(opaquePortsNs); err != nil {
+			testutil.AnnotatedFatal(t, "failed to deploy applications", err)
+		}
+		waitForAppDeploymentReady(t, opaquePortsNs)
 
-			if err != nil {
-				testutil.AnnotatedFatalf(t, "unexpected metric output", "unexpected metric output: %v", err)
-			}
-		})
-
-		t.Run("expect inbound TCP connection metric with no TLS identity for traffic between meshed and unmeshed opaque service", func(t *testing.T) {
-			// Slow cooker is meshed, should have valid outbound TCP metric, valid
-			// inbound TCP metric and no HTTP metric.
-			err := TestHelper.RetryFor(30*time.Second, func() error {
-				pods, err := TestHelper.GetPods(ctx, opaquePortsNs,
-					map[string]string{"app": opaqueUnmeshedSvcSC})
-				if err != nil {
-					return fmt.Errorf("error getting pods\n%w", err)
-				}
-				metrics, err := getPodMetrics(pods[0], opaquePortsNs)
-				if err != nil {
-					return fmt.Errorf("error getting metrics for pod\n%w", err)
-				}
+		tmplArgs, err := templateArgsPodIP(ctx, opaquePortsNs)
+		if err != nil {
+			testutil.AnnotatedFatal(t, "failed to fetch service IPs", err)
```

Comment:
```suggestion
			testutil.AnnotatedFatal(t, "failed to fetch Pod IPs", err)
```

Reply:
Fixed in https://github.com/linkerd/linkerd2/pull/8225/commits/a2f7944f42a2151185fe3109ed456f3b05df9449 I went for lower case, because it seems more natural to me in this case.